examples/pennylane_run_doubly_stochastic.py

```diff
@@ -0,0 +1,312 @@
+#         ~~a_{i,j} = \frac{1}{4}\text{tr}[(\sigma_i\otimes \sigma_j )H].
+#
+# Applying this, we can see that
+#
+# .. math::
+#
+#     H = 4  + 2I\otimes X + 4I \otimes Z - X\otimes X + 5 Y\otimes Y + 2Z\otimes X.
+#
+# To perform "doubly stochastic" gradient descent, we simply apply the stochastic
+# gradient descent approach from above, but in addition also uniformly sample
+# a subset of the terms for the Hamiltonian expectation at each optimization step.
+# This inserts another element of stochasticity into the system---all the while
+# convergence continues to be guaranteed!
+#
+# Let's create a QNode that randomly samples three terms from the above
+# Hamiltonian as the observable to be measured.
+
+I = np.identity(2)
+X = np.array([[0, 1], [1, 0]])
+Y = np.array([[0, -1j], [1j, 0]])
+Z = np.array([[1, 0], [0, -1]])
+
+terms = np.array(
+    [2 * np.kron(I, X), 4 * np.kron(I, Z), -np.kron(X, X), 5 * np.kron(Y, Y), 2 * np.kron(Z, X)]
+)
+
+
+@qml.qnode(dev_SGD)
+def sampled_terms(params, A=None):
+    StronglyEntanglingLayers(*params, wires=[0, 1])
+    return expval(qml.Hermitian(A, wires=[0, 1]))
+
+
+def loss(params):
+    n = np.random.choice(np.arange(5), size=3, replace=False)
+    A = np.sum(terms[n], axis=0)
+    return 4 + (5 / 3) * sampled_terms(params, A=A)
+
+
+##############################################################################
+# Optimizing the circuit using gradient descent via the parameter-shift rule:
+
+dev_SGD.shots = 100
+cost = []
+params = init_params
+opt = qml.GradientDescentOptimizer(0.005)
+
+for _ in range(250):
+    cost.append(loss(params))
+    params = opt.step(loss, params)
+
+##############################################################################
+# Plotting the cost versus optimization step:
+
+
+def moving_average(data, n=3):
+    ret = np.cumsum(data, dtype=np.float64)
+    ret[n:] = ret[n:] - ret[:-n]
+    return ret[n - 1 :] / n
+
+
+average = np.vstack([np.arange(25, 200), moving_average(cost, n=50)[:-26]])
+
+plt.plot(cost_GD, label="Vanilla gradient descent")
+plt.plot(cost, ".", label="Doubly stochastic gradient descent")
+plt.plot(average[0], average[1], "--", label="Doubly stochastic gradient descent moving average")
+plt.hlines(min_energy, 0, 200, linestyles=":", label="Ground state energy")
+
+plt.ylabel("Cost function value")
+plt.xlabel("Optimization steps")
+plt.xlim(-2, 200)
+plt.legend()
+plt.show()
+
+##############################################################################
+# Finally, verifying that the doubly stochastic gradient descent optimization
+# correctly provides the ground state energy when evaluated for a larger
+# number of shots:
+
+print("Doubly stochastic gradient descent min energy = ", qnode_GD(params))
+
+##############################################################################
+# References
+# ----------
+#
+# 1. Ryan Sweke, Frederik Wilde, Johannes Meyer, Maria Schuld, Paul K. Fährmann,

```

Comment:
```suggestion
# 1. Ryan Sweke, Frederik Wilde, Johannes Jakob Meyer, Maria Schuld, Paul K. Fährmann,
```
I forgot my middle name on the arXiv submission, but we'll change it for the next version.